src/java/org/apache/cassandra/service/paxos/PaxosState.java

```diff
@@ -633,28 +640,36 @@ else if (isAfter(ballot, latestWriteOrLowBound))
     /**
      * Record an acceptance of the proposal if there is no newer promise; otherwise inform the caller of the newer ballot
      */
-    public Ballot acceptIfLatest(Proposal proposal)
+    public AcceptResult acceptIfLatest(Proposal proposal, boolean isForRepair)
     {
         if (paxosStatePurging() == legacy && !(proposal instanceof AcceptedWithTTL))
             proposal = AcceptedWithTTL.withDefaultTTL(proposal);
 
         // state.promised can be null, because it is invalidated by committed;
         // we may also have accepted a newer proposal than we promised, so we confirm that we are the absolute newest
         // (or that we have the exact same ballot as our promise, which is the typical case)
+        boolean shouldRejectDueToConsensusMigration;
         Snapshot before, after;
         while (true)
         {
             Snapshot realBefore = current;
             before = realBefore.removeExpired((int)proposal.ballot.unix(SECONDS));
             Ballot latest = before.latestWitnessedOrLowBound();
+            if (isForRepair)
+                shouldRejectDueToConsensusMigration = false;
+            else
+                shouldRejectDueToConsensusMigration = ConsensusRequestRouter.instance
+                                                          .isKeyInMigratingOrMigratedRangeDuringPaxosAccept(proposal.update.metadata().id,
+                                                                                                            proposal.update.partitionKey());
             if (!proposal.isSameOrAfter(latest))
             {
                 Tracing.trace("Rejecting proposal {}; latest is now {}", proposal.ballot, latest);
-                return latest;
+                return new AcceptResult(latest, shouldRejectDueToConsensusMigration);
```

Comment:
we can use singletons for each `shouldRejectDueToConsensusMigration` value

Reply:
Added a singleton for success and retry on new protocol.